docs/modules/postgres.md

```diff
@@ -0,0 +1,97 @@
+- `PostgresContainerOption`, a variad argument for passing options.
+
+## Container Options
+
+When starting the Postgres container, you can pass options in a variadic way to configure it.
+
+!!!tip
+    You can find all the available configuration and environment variables for the Postgres Docker image on [Docker Hub](https://hub.docker.com/_/postgres).
+
+### Image
+
+If you need to set a different Postgres Docker image, you can use `WithImage` with a valid Docker image
+for Postgres. E.g. `WithImage("docker.io/postgres:9.6")`.
+
+### Initial Database
+
+If you need to set a different database, and its credentials, you can use the `WithInitialDatabase`.
+
+<!--codeinclude-->
+[Set Initial database](../../modules/postgres/postgres_test.go) inside_block:withInitialDatabase
+<!--/codeinclude-->
+
+### Init Scripts
+
+If you would like to do additional initialization in the Postgres container, add one or more `*.sql`, `*.sql.gz`, or `*.sh` scripts to the container request.
+Those files will be copied after the container is created but before it's started under `/docker-entrypoint-initdb.d`. According to Postgres Docker image,
+it will run any `*.sql` files, run any executable `*.sh` scripts, and source any non-executable `*.sh` scripts found in that directory to do further
+initialization before starting the service.
+
+<!--codeinclude-->
+[Include init scripts](../../modules/postgres/postgres_test.go) inside_block:withInitScripts
+<!--/codeinclude-->
+
+<!--codeinclude-->
+[Init script content](../../modules/postgres/testresources/init-user-db.sh)
+<!--/codeinclude-->
+
+### InitDB arguments
+
+This optional function can be used to send arguments to postgres initdb. The value is a space separated string of arguments as postgres initdb would expect them.
+This is useful for adding functionality like data page checksums: `WithInitDBArgs("--data-checksums")`.
+
+### Wait Strategies
+
+Given you could need to wait for different conditions, in particular using a wait.ForSQL strategy,
+the Postgres container exposes a `WithWaitStrategy` option to set a custom wait strategy.
+
+<!--codeinclude-->
+[Set Wait Strategy](../../modules/postgres/postgres_test.go) inside_block:withWaitStrategy
+<!--/codeinclude-->
+
+## Container Methods
+
+### ConnectionString
+
+This method returns the connection string to connect to the Postgres container, using the default `5432` port.
+
+<!--codeinclude-->
+[Get connection string](../../modules/postgres/postgres_test.go) inside_block:connectionString
+<!--/codeinclude-->
+
+## Postgres variants
+
+It's possible to use the Postgres container with Timescale or Postgis, to name a few. You simply need to update the image name and the wait strategy.
+
+<!--codeinclude-->
+[Image for Timescale](../../modules/postgres/postgres_test.go) inside_block:timescale
+<!--/codeinclude-->
+
+<!--codeinclude-->
+[Image for Postgis](../../modules/postgres/postgres_test.go) inside_block:postgis
+<!--/codeinclude-->
```

Comment:
👍🏽

Reply:
Instead of creating specialised containers for Timescale and Postgis, I decided to create tests for it. They could be Example, otoh 🤔